java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -819,22 +812,24 @@ public SymbolicValue wrappedValue() {
     }
   }
 
-  private SymbolicValue getCaughtException(Type caughtType) {
-    SymbolicValue sv = null;
+  private SymbolicValue handleException(Type caughtType) {
+    SymbolicValue.ExceptionalSymbolicValue sv = null;
     Type exceptionType = null;
     // FIXME SONARJAVA-2069 every path conducting to a catch block should have an exceptional symbolic value on top of the stack
     if (programState.peekValue() instanceof SymbolicValue.ExceptionalSymbolicValue) {
+      // clear the stack by consuming the exception
```

Comment:
Actually, stack should be really cleared by calling `programState.clearStack()` not just popping one SV. See https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-6.html#jvms-6.5.athrow  . This might be important for being able to hit cached node in EG. Also it should be cleared regardless if the type is known or not (so perhaps should be done in `executeVariable` together with setting constraint)